### PR TITLE
v1.10 backports 2021-08-10

### PR DESCRIPTION
**Pull request by gandro**
* #16918 -- operator: misc. refactoring and code removal (@aanm)
 * ~#17000 -- config: Fix incorrect packet path with IPsec and endpoint routes (@pchaigno)~
 * #16947 -- install: Fix README links to getting started guides (@joestringer)
 * #16991 -- policy: Fix `cilium policy trace` output when only deny rules are applied (@chez-shanpu)
 * #17014 -- docs: Fix missing quote in gcloud command for GKE (@christarazi)
 * #17035 -- node-neigh: Wait instead of sleeping in unit tests (@aanm)
 * #16656 -- Add Bidirectional mount in Cilium DaemonSet (@aanm)
 * #17027 -- Include '*.mesh.cilium.io' in 'clustermesh-apiserver''s server certificate. (@bmcustodio)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16918 16947 16991 17014 17035 16656 17027; do contrib/backporting/set-labels.py $pr done 1.10; done
```

### CODE REVIEW

**Comment by gandro** [comment]
@aanm Please review bfa7f970c26fb199be2425f4aa843e8fe54da88b carefully, I had to resolve some conflicts due to recent Helm refactors.

**Comment by gandro** [comment]
test-backport-1.10

**Comment by pchaigno** [comment]
We've reverted https://github.com/cilium/cilium/pull/17000 because it was breaking IPsec connectivity tests on AKS, so we probably shouldn't backport it. I've just removed the labels from that PR. Sorry I wasn't to anticipate that :-(

**Comment by gandro** [comment]
> We've reverted #17000 because it was breaking IPsec connectivity tests on AKS, so we probably shouldn't backport it. I've just removed the labels from that PR. Sorry I wasn't faster to anticipate that :-(

Dropped!

**Review by pchaigno — APPROVED** [review]
> policy: Fix cilium policy trace output when only deny rules are applied #16991 -- policy: Fix cilium policy trace output when only deny rules are applied (@chez-shanpu)

:+1: for that PR.

**Review by bmcustodio — APPROVED** [review]

File: install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml [hunk 1]
```diff
@@ -425,11 +425,6 @@ spec:
           valueFrom:
             configMapKeyRef:
               key: clean-cilium-bpf-state
-              name: cilium-config
-              optional: true
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml]: This should be kept L425-L429

File: install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml [hunk 2]
```diff
@@ -425,11 +425,6 @@ spec:
...
-        - name: CILIUM_WAIT_BPF_MOUNT
-          valueFrom:
-            configMapKeyRef:
               key: wait-bpf-mount
               name: cilium-config
               optional: true
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml:429]: This should be removed L430-L435

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml (hunk 1 above)
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml]: I did exactly what the original commit did, `diff` just displays it weirdly. But I removed the `key: wait-bpf-mount` line now, which means https://github.com/cilium/cilium/pull/17123 will not have to be backported.

**Review by christarazi — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-backport-1.10

**Comment by gandro** [comment]
Seems like the flake is #12511 ( https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-net-next/1230/)

**Comment by gandro** [comment]
test-1.16-netnext

**Comment by aanm** [comment]
ci-awscni-1.10

**Comment by aanm** [comment]
ci-aks-1.10

**Comment by aanm** [comment]
ci-eks-1.10

**Comment by aanm** [comment]
ci-gke-1.10

**Comment by aanm** [comment]
ci-multicluster-1.10

**Comment by gandro** [comment]
The the IPSec+L7 part of ci-eks-1.10 is currently known to be broken: #17139. The rest is not applicable to v1.10, marking ready to merge.

**Pull request merged**